geotiepoints/viiinterpolator.py

```diff
@@ -64,39 +65,57 @@ def tie_points_interpolation(data_on_tie_points, scan_alt_tie_points, tie_points
 
     # Compute the dimensions of the pixel points array across and along track
     n_pixel_act = (n_tie_act - 1) * tie_points_factor
-    n_pixel_alt = (n_tie_alt - 1) * tie_points_factor
+    n_pixel_alt = (scan_alt_tie_points - 1) * tie_points_factor * n_scans
 
     # Create the grids used for interpolation across the track
     tie_grid_act = da.arange(0, n_pixel_act + 1, tie_points_factor)
     pixels_grid_act = da.arange(0, n_pixel_act)
 
-    # Create the grids used for the interpolation along the track (must not include the spurious points between scans)
-    tie_grid_alt = da.arange(0, n_pixel_alt + 1, tie_points_factor)
-    n_pixel_alt_per_scan = (scan_alt_tie_points - 1) * tie_points_factor
-    pixel_grid_alt = []
-    for j_scan in range(n_scans):
-        start_index_scan = j_scan * scan_alt_tie_points * tie_points_factor
-        pixel_grid_alt.append(da.arange(start_index_scan, start_index_scan + n_pixel_alt_per_scan))
-    pixel_grid_alt = da.concatenate(pixel_grid_alt)
+    # Create the grids used for the interpolation along the track for the current scan
+    n_pixel_alt_scan = (scan_alt_tie_points - 1) * tie_points_factor
+    tie_grid_alt_scan = da.arange(0, n_pixel_alt_scan + 1, tie_points_factor)
+    pixels_grid_alt_scan = da.arange(0, n_pixel_alt_scan)
 
-    # Loop on all arrays
     data_on_pixel_points = []
+
+    # loop over granules
     for data in data_on_tie_points:
 
-        if data.shape != (n_tie_act, n_tie_alt) or data.dims != (dim_act, dim_alt):
-            raise ValueError("The dimensions of the arrays are not consistent")
+        # create an xarray to hold the data on pixel grid (note the coords get renamed later so keep the old names)
+        rads = da.zeros((n_pixel_alt, n_pixel_act))
+        pix_act = da.zeros(n_pixel_act)
+        pix_alt = da.zeros(n_pixel_alt)
+
+        data_on_pixel_points_granule = \
+            xr.DataArray(rads, dims=['num_tie_points_alt', 'num_tie_points_act'],
+                         coords={'num_tie_points_alt': pix_alt, 'num_tie_points_act': pix_act})
+        data_on_pixel_points_granule.attrs = combine_metadata(data)
+
+        # loop over scans
+        for j_scan in range(n_scans):
+            index_tie_points_start = j_scan * scan_alt_tie_points
+            index_tie_points_end = (j_scan * scan_alt_tie_points) + scan_alt_tie_points
+            index_pixel_start = j_scan * (scan_alt_tie_points - 1) * tie_points_factor
+            index_pixel_end = (j_scan * (scan_alt_tie_points - 1) * tie_points_factor) + \
+                              (scan_alt_tie_points - 1) * tie_points_factor
```

Comment:
Do I understand correctly that this has been moved inside the granule loop and is now being repeate for every array?
What is the performance hit of this?
Also have you tried vectorizing this loop?

Reply:
There is over a factor of 10 hit on performance so I have reverted to the original methodology which interpolates the granule as a whole, although not strictly accurate. We will investigate the impact of this on geolocation accuracy further but for now I think its the most practical solution.